dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/DefaultDataValueSetService.java

```diff
@@ -1336,19 +1336,24 @@ private void handleComplete( DataSet dataSet, Date completeDate, Period period,
         CompleteDataSetRegistration completeAlready = registrationService
             .getCompleteDataSetRegistration( dataSet, period, orgUnit, attributeOptionCombo );
 
-        String username = currentUserService.getCurrentUsername();
-
         if ( completeAlready != null )
         {
-            completeAlready.setStoredBy( username );
+            //The method is called only via /dataSets endpoint which cannot un-complete the data set.
```

Comment:
What method? handleComplete?

If you feel this comment is needed, could you simplify it, and maybe use /* */ for multiline comment? By simplify, I mean instead of writing like you are uncertain, write more decisive (For example: DataSet is completed at this point, so we do this and that...)